[FLINK-16746] Deprecate legacy heap memory option for JM and expose the new ones in docs

azagrebin opened the pull request:
## What is the purpose of the change

Add default 'jobmanager.memory.process.size: 1472m' to flink-conf.yaml.

Deprecate legacy heap options: `jobmanager.heap.size` (and update deprecation of `jobmanager.heap.mb`) to use `jobmanager.memory.heap.size` for standalone and `jobmanager.memory.process.size` for containerized (Yarn/K8s) environments.

Expose the new JM memory options in user docs.

## Verifying this change

Trivial change. Check docs.

azagrebin commented:
The PR is based on #11577, #11545 and  #11675 atm.
Only two last commits are to review.
cc @xintongsong

xintongsong commented on flink-dist/src/main/resources/flink-conf.yaml:
```diff
@@ -37,9 +37,11 @@ jobmanager.rpc.address: localhost
+# Note this accounts for all memory usage within the JobManager process, including JVM metaspace and other overhead.
 
-jobmanager.heap.size: 1024m
+jobmanager.memory.process.size: 1472m
```
Could you explain why is the default 1472m?

xintongsong commented on flink-jepsen/src/jepsen/flink/db.clj:
```diff
@@ -41,7 +41,7 @@
    :high-availability.zookeeper.quorum    (zookeeper-quorum test)
    :high-availability.storageDir          "hdfs:///flink/ha"
-   :jobmanager.heap.size                  "2048m"
+   :jobmanager.memory.process.size        "2496m"
```
Same here? Why is the value change "2048 -> 2496"?

azagrebin replied:
same here, it shoud be `2702m`. Though, as discussed with @GJL, it does not matter too much in this case.

azagrebin commented:
Thanks for the review @xintongsong 
I addressed the comments.

The last thing:
> There are links in the mem_setup.md and mem_setup.zh.md pointing to config.html#jobmanager-heap-size, which I believe is removed.
we can address in [FLINK-16946](https://issues.apache.org/jira/browse/FLINK-16946). It is already planned to do so in [the doc draft](https://docs.google.com/document/d/1T7AFm2TbVcb3YkPlSiIFMWyZkxwVbSx4jpLbvdRLi3I/edit#heading=h.qgasvihwxdaf) when the JM chapter becomes available. For now, this link will just go to the `config` page which should be ok.
